source/Calamari/Deployment/PackageRetention/Caching/PercentFreeDiskSpacePackageCleaner.cs

```diff
@@ -32,6 +32,7 @@ public PercentFreeDiskSpacePackageCleaner(ICalamariFileSystem fileSystem, ISortJ
 
         public IEnumerable<PackageIdentity> GetPackagesToRemove(IEnumerable<JournalEntry> journalEntries)
         {
+            var journalEntriesCanBeRemoved = journalEntries.Where(entry => !entry.HasLock());
```

Comment:
The sorting implementation respects locked packages: https://github.com/OctopusDeploy/Calamari/blob/master/source/Calamari/Deployment/PackageRetention/Caching/LeastFrequentlyUsedWithAgingSort.cs#L24C17-L24C25